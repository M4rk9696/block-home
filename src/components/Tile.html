<div class="page-tile" on:click="fire('select', tile.url)" >
  <div class="page-tile-block" style="background: {tile.color}">
    <div class="top-buttons">
      <button class="close" on:click="fire('removeTile')">
        {#if tile.url !== ''}
          &#10006;
        {/if}
      </button>
    </div>
    <span class="page-tile-char">{tile.character}</span>
    <span class="page-tile-title">{tile.title}</span>
  </div>
</div>

<style>
  .page-tile-char {
    display: block;
    margin: 5px auto;
    padding: 2px;
    font-size: 2rem;
    text-align: center;
    text-transform: capitalize;
  }
  .page-tile-title {
    display: block;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    text-align: center;
    padding: 20px 0;
  }
  .page-tile-block {
    padding: 10px;
    height: 130px;
    border-radius: 15px;
    transition: transform 100ms ease;
  }
  .page-tile-block:hover {
    transform: scale(1.2);
  }
  .page-tile {
    color: #000;
  }
  .page-tile:hover {
    color: #000;
    text-decoration: none;
  }
  .page-tile:visited {
    color: #000;
  }
</style>
