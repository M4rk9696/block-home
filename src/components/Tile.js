import React from 'react'
import styled from 'styled-components'
import Close from './Close'

const Tile = ({ url, character, color, title }) => {
  return (
    <>
      <PageTile>
        <PageTileBlock color={color}>
          <div className="top-buttons">
            <Close>
              <If condition={url !== ''}>&#10006;</If>
            </Close>
          </div>
          <PageTileChar>{character}</PageTileChar>
          <PageTileTitle>{title}</PageTileTitle>
        </PageTileBlock>
      </PageTile>
    </>
  )
}

const PageTileChar = styled.span`
  display: block;
  margin: 5px auto;
  padding: 2px;
  font-size: 2rem;
  text-align: center;
  text-transform: capitalize;
`

const PageTileTitle = styled.span`
  display: block;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  text-align: center;
  padding: 20px 0;
`

const PageTileBlock = styled.div`
  margin: 5px;
  padding: 10px;
  height: 130px;
  max-width: 200px;
  border-radius: 15px;
  background: ${props => props.color};
  transition: transform 100ms ease;

  &:hover {
    transform: scale(1.2);
  }
`

const PageTile = styled.div`
  flex-basis: 150px;
  width: fit-content;
  width: -moz-fit-content;
  height: fit-content;
  height: -moz-fit-content;
  color: #000;

  &:hover {
    color: #000;
    text-decoration: none;
  }
  &e:visited {
    color: #000;
  }
`

export default Tile
