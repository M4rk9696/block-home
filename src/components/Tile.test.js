import React from 'react'
import {render} from '@testing-library/react'
import Tile from './Tile'

describe('should render Tile', () => {
  it('should be able to mount the component', () => {
    const {container} = render(<Tile
      url="dummy-url"
      color="#f3f3f3"
      character="D"
      title="Dummy" />)

    expect(container).toMatchSnapshot()
  })
})
