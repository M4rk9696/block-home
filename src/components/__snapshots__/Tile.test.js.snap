// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Tile should be able to mount the component 1`] = `
<div>
  <div
    class="sc-ifAKCX dEMnFe"
  >
    <div
      class="sc-bxivhb WGaQA"
      color="#f3f3f3"
    >
      <div
        class="top-buttons"
      >
        <button
          class="sc-bdVaJa iibjPH"
        >
          âœ–
        </button>
      </div>
      <span
        class="sc-bwzfXH iyaRUM"
      >
        D
      </span>
      <span
        class="sc-htpNat fPitBk"
      >
        Dummy
      </span>
    </div>
  </div>
</div>
`;
